### ISSUE TITLE
Feature request: Support for drag handlers

### ISSUE THREAD

**Issue by devon** [baumths/flutter_tree_view#74]
It would be nice to add a drag handler to TreeDraggable.  That way, we can easily scroll the tree and use the handler to drag nodes.

<img width="368" alt="Screen Shot 2023-10-23 at 8 34 04 AM" src="https://github.com/baumths/flutter_tree_view/assets/8176/9dc3e442-25f8-40cd-a018-d2d76cf8a04f">

**Comment by baumths** [comment]
Hey @devon, thank you for using this package!

A drag handle approach should be doable already. You'd have to move the `TreeDraggable` from around your tree tile down the widget tree to wrap your drag handle icon.

I intend to update the example app to be more mobile friendly (i.e., long press to drag or a drag handle as requested) when I get some spare time.

**Comment by devon** [comment]
> Hey @devon, thank you for using this package!
> 
> A drag handle approach should be doable already. You'd have to move the `TreeDraggable` from around your tree tile down the widget tree to wrap your drag handle icon.
> 
> I intend to update the example app to be more mobile friendly (i.e., long press to drag or a drag handle as requested) when I get some spare time.

The mobile friendly example would be great. Thank you very much. I am starting to use this widget in a new app and it works very well. Love it.

For mobile drag-and-drop outlining (tree), the best app I have tried so far is Cloud Outliner. It supports dragging and dropping multiple items at once.

**Comment by urvashik-7span** [comment]
I am facing the same issue for scrolling and dragging. Any update on this?

**Comment by baumths** [comment]
Hey folks, sorry for taking so long to look into this. I just pushed a fix to the demo app which is now requiring a long press to start dragging on mobile platforms, this should fix the scrolling issue.

About the drag handle, I wrote a quick and unpolished version for demonstration purposes, I've tested it using the current DnD example from the demo app. It's currently a bit wonky on RTL directionality e.g., the feedback must be translated in `TreeDraggable.dragAnchorStrategy` by the width of the TreeTile's content. Take a look:
